app/timecode/views.py

```diff
@@ -0,0 +1,216 @@
+import os
+import re
+import difflib
+import tempfile
+import urllib
+import time
+from pprint import pprint
+
+from flask import Blueprint, current_app, request
+from flask.helpers import make_response
+import webvtt
+
+from app.doc2html.conversionFunctions import docx_to_text
+from app.doc2html.views import process_docx_uid
+
+blueprint = Blueprint('time_code', __name__)
+
+FIX_DERIVE_ZERO = 10 ** -5
+
+vtt_q = """
+SELECT f.uid from files f 
+INNER JOIN content_units cu ON cu.id = f.content_unit_id
+WHERE cu.uid = '%s' 
+AND f.properties->>'insert_type' = 'subtitles'
+AND f.language = '%s'
+"""
+file_q = """
+        select f.* from files f
+        INNER JOIN content_units cu ON cu.id = f.content_unit_id
+        WHERE cu.uid = '%s'
+        AND f.language = '%s'
+        """
+
+
+@blueprint.route('/time_code')
+def timecode():
```

Comment:
Please add comments and descriptions to the whole file.

Reply:
added